coprocess_python.go

```diff
@@ -295,20 +296,20 @@ func NewCoProcessDispatcher() (coprocess.Dispatcher, error) {
 		paths = append(paths, v)
 	}
 
-	PythonSetEnv(paths...)
-
-	PythonInit()
-	PythonLoadDispatcher()
-
-	dispatcher, err := PythonNewDispatcher(middlewarePath, eventHandlerPath, bundlePaths)
-
-	C.PyEval_ReleaseLock()
-
-	if err != nil {
-		log.WithFields(logrus.Fields{
-			"prefix": "coprocess",
-		}).Error(err)
-	}
-
+	go func() {
+		runtime.LockOSThread()
+		defer runtime.UnlockOSThread()
+		PythonSetEnv(paths...)
+		PythonInit()
+		PythonLoadDispatcher()
+		dispatcher, err = PythonNewDispatcher(middlewarePath, eventHandlerPath, bundlePaths)
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"prefix": "coprocess",
+			}).Error(err)
+		}
+		pythonInitDone <- true
+	}()
+	<-pythonInitDone
```

Comment:
You probably want to make `pythonInitDone` an error channel, so that the error can be returned or otherwise handled.